src/main/java/com/urlparser/model/Links.java

```diff
@@ -0,0 +1,8 @@
+package com.urlparser.model;
+
+import java.net.URL;
+
+public interface Links extends Iterable<URL> {
```

Comment:
@undsett can we make it generic? Like `Links<T> extends Iterable<T>`
This way we may have links not only as `URL` but also as `String`, which is important since `Link<URL>` implementation depends on `Links<String>` - Raw list of strings taken from `a[href]`.
This means we can use decoration in order to decorate `Links<Strings>`  and implement `Links<URL>`, can we do this?